walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,123 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
 
 public final class FindMeetingQuery {
+  
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    final Collection<String> requestAttendees = request.getAttendees();
+    final long requestDuration = request.getDuration();
+    ArrayList<TimeRange> eventList = createFlattenedEventList(events, requestAttendees);
+
+    if(requestDuration > TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    if(requestAttendees.isEmpty() || eventList.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+    return findMeetings(eventList, requestDuration);
+  }
+  
+   private ArrayList<TimeRange> createEventList(Collection<Event> events, Collection<String> requestAttendees) {
+      ArrayList<TimeRange> eventList = new ArrayList<>();
+      for(Event event: events) {
+          Set<String> attendees = event.getAttendees();
```

Comment:
You could probably line this variable on the next line, like `for (String attendee : event.getAttendees())`. Creating separate variables is only useful when (1) it's called many times (here it's only called once), or (2) if the logic is very complicated so a better named variable is more descriptive (`event.getAttendees()` is pretty straightforward - perhaps maybe even more straightforward than just `attendees`).